src/overlays/actors/ovl_En_Torch2/z_en_torch2.c

```diff
@@ -27,21 +66,715 @@ const ActorInit En_Torch2_InitVars = {
+                             * invincibility timer indicates this state, and results in the bug noted below.
+                             */
+                            if (!this->swordState && (sCounterState == 0) && (player->invincibilityTimer == 0) &&
+                                (player->swordAnimation == STAB_1H) && (this->actor.xzDistFromLink <= 85.0f) &&
+                                func_80033A84(globalCtx, &this->actor)) {
+
+                                sStickTilt = 0.0f;
+                                sSwordJumpState = 1;
+                                player->stateFlags3 |= 4;
+                                this->actor.flags &= ~1;
+                                sSwordJumpTimer = 27;
+                                player->swordState = 0;
+                                player->linearVelocity = 0.0f;
+                                this->invincibilityTimer = -7;
+                                this->linearVelocity = 0.0f;
+                                player->skelAnime.animCurrentFrame = 2.0f;
+                                func_800A3BC0(globalCtx, &player->skelAnime);
+                                sHoldShieldTimer = 0;
+                                input->cur.button = BTN_A;
+                            } else {
+                                EnTorch2_Backflip(this, input, &this->actor);
+                            }
+                        } else {
+                            /*
+                             * Handles reactions to all other sword attacks
+                             */
+                            sStickAngle = this->actor.yawTowardsLink; // Not loaded into pointer
+                            input->cur.button = BTN_B;
+                            
+                            if (player->swordAnimation <= FORWARD_COMBO_2H) {
+                                sStickTilt = 0.0f;
+                            } else if (player->swordAnimation <= RIGHT_COMBO_2H) {
+                                sStickTilt = 127.0f;
+                                sStickAngle += sStickAngle; // Not loaded from pointer
+                            } else if (player->swordAnimation <= LEFT_COMBO_2H){
+                                sStickTilt = 127.0f;
+                                sStickAngle += -0x4000; // Not loaded from pointer
+                            } else if (player->swordAnimation <= HAMMER_SIDE) {
+                                input->cur.button = BTN_R;
+                            } else if (player->swordAnimation <= BIG_SPIN_2H) {
+                                EnTorch2_Backflip(this, input, &this->actor);
+                            } else {
+                                EnTorch2_Backflip(this, input, &this->actor);
+                            }
+                            if (!(input->cur.button & (BTN_A | BTN_R)) && !this->swordState && player->swordState) {
+                                sCounterState = 1;
```

Comment:
is sword state a bool? had the impression from earlier that it wasnt.
if it is then itll have to be made consistent

Reply:
It is not, even though it's essentially use that way here, because there's a slight distinction between `-1` and `1`. Incidentally, other actors with swords (like EnGeldB) also do this.

Reply:
ok, then yeah i believe there are some comparisons like `if (!this->swordState)` that need to be changed to non bool

Reply:
and they have been.

Reply:
there are still some left that need to change

Reply:
There are? I thought I got them all.